Allow passing options directly to server instead of setting global env

hauleth opened the pull request:
Closes #27

hauleth committed:
feat: allow passing configuration directly to server

ferd commented:
This PR does not correctly implement https://github.com/ferd/vmstats/issues/27 since it won't let you have multiple servers, but I think the changeset is fine otherwise.

ferd commented on src/vmstats_server.erl:
```diff
@@ -22,14 +29,19 @@
+    Interval = proplists:get_value(interval, Options, 1000),
+    KeySeparator = proplists:get_value(key_separator, Options, $.),
+    SchedTimeEnabled = proplists:get_value(sched_time, Options, true),
+    MemoryMetrics = proplists:get_value(memory_metrics, Options,
```
I'd use lists:keyfind with a hard pattern match -- the default values are provided by the .app's env tuple (as you could see before, no defaults were given in code).

Otherwise this implicitly creates a new layer of default options in the code that backstops the .app file, but we shouldn't introduce that unless we really need to

hauleth committed:
feat: do not register server under name

This will allow running multiple vmstats servers in one VM without name
collisions.  It can be useful for integration.

edgurgel commented:
Thanks for this PR, @hauleth! It will be very useful :)

I would love to start `vmstats_server` just when my sink is ready to receive data. 

Hey @ferd, would you mind releasing a new version with this change included? Thanks for this great library!

ferd commented:
Took a while, but it's done.

edgurgel commented:
Thanks, @ferd 🎉
